type hints need to be updated for lark.Lark's postlex parameter

**Describe the bug**

Mypy complain about the following code:
```py
from pathlib import Path
from typing import List

import lark
import lark.indenter

here = Path(__file__).parent


class Indenter(lark.indenter.Indenter):
    NL_type = "_NL"
    OPEN_PAREN_types: List[str] = []
    CLOSE_PAREN_types: List[str] = []
    INDENT_type = "_INDENT"
    DEDENT_type = "_DEDENT"
    tab_len = 4


grammar = lark.Lark(
    here.joinpath("grammar.lark").read_text(), parser="lalr", postlex=Indenter()
)
```
And it complains:

```
foo/parse.py:20: error: Argument "postlex" to "Lark" has incompatible type "Indenter"; expected "Optional[PostLex]"
Found 1 error in 1 file (checked 1 source file)
```
I'm copy-pasting [the docs](https://lark-parser.readthedocs.io/en/latest/examples/indented_tree.html) (with some edits). I believe this is a type stub problem.

---

(Poetry) Lockfile: https://gist.github.com/ThatXliner/e311980a936ef568e2d9caeb0fb5d2a7
Python: 3.6.13

**To Reproduce**

See above

Provide a short script that reproduces the erroneous behavior.

If that is impossible, provide clear steps to reproduce the behavior.

--- a/lark-stubs/grammar.pyi
+++ b/lark-stubs/grammar.pyi
@@ -7,3 +7,8 @@ class RuleOptions:
     priority: int
     template_source: Optional[str]
     empty_indices: Tuple[bool, ...]
+
+
+class Symbol:
+    name: str
+    is_term: bool

--- a/lark-stubs/indenter.pyi
+++ b/lark-stubs/indenter.pyi
@@ -3,9 +3,10 @@
 from typing import Tuple, List, Iterator, Optional
 from abc import ABC, abstractmethod
 from .lexer import Token
+from .lark import PostLex
 
 
-class Indenter(ABC):
+class Indenter(PostLex, ABC):
     paren_level: Optional[int]
     indent_level: Optional[List[int]]
 
@@ -13,13 +14,6 @@ class Indenter(ABC):
         ...
 
     def handle_NL(self, token: Token) -> Iterator[Token]:
-        ...
-
-    def process(self, stream: Iterator[Token]) -> Iterator[Token]:
-        ...
-
-    @property
-    def always_accept(self) -> Tuple[str]:
         ...
 
     @property

--- a/lark-stubs/lark.pyi
+++ b/lark-stubs/lark.pyi
@@ -65,7 +65,7 @@ class Lark:
         grammar: Union[Grammar, str, IO[str]],
         *,
         start: Union[None, str, List[str]] = "start",
-        parser: Literal["earley", "lalr", "cyk"] = "auto",
+        parser: Literal["earley", "lalr", "cyk", "auto"] = "auto",
         lexer: Union[Literal["auto", "standard", "contextual", "dynamic", "dynamic_complete"], Type[Lexer]] = "auto",
         transformer: Optional[Transformer] = None,
         postlex: Optional[PostLex] = None,

--- a/lark-stubs/load_grammar.pyi
+++ b/lark-stubs/load_grammar.pyi
@@ -1,8 +1,8 @@
 from typing import List, Tuple, Union, Callable, Dict, Optional
 
-from lark import Tree
-from lark.grammar import RuleOptions
-from lark.exceptions import UnexpectedInput
+from .tree import Tree
+from .grammar import RuleOptions
+from .exceptions import UnexpectedInput
 
 
 class Grammar:

--- a/lark-stubs/reconstruct.pyi
+++ b/lark-stubs/reconstruct.pyi
@@ -11,8 +11,7 @@ from .lexer import TerminalDef
 
 class WriteTokensTransformer(Transformer_InPlace):
 
-    def __init__(self, tokens: Dict[str, TerminalDef], Dict[str, Callable[[Symbol], str]] = ...):
-        ...
+    def __init__(self, tokens: Dict[str, TerminalDef], term_subs: Dict[str, Callable[[Symbol], str]] = ...): ...
 
 
 class MatchTree(Tree):

--- a/lark/indenter.py
+++ b/lark/indenter.py
@@ -1,13 +1,14 @@
 "Provides Indentation services for languages with indentation similar to Python"
 
 from .exceptions import LarkError
+from .lark import PostLex
 from .lexer import Token
 
 ###{standalone
 class DedentError(LarkError):
     pass
 
-class Indenter:
+class Indenter(PostLex):
     def __init__(self):
         self.paren_level = None
         self.indent_level = None

--- a/lark/lark.py
+++ b/lark/lark.py
@@ -1,4 +1,6 @@
 from __future__ import absolute_import
+
+
 from lark.exceptions import ConfigurationError, assert_config
 
 import sys, os, pickle, hashlib
@@ -6,7 +8,7 @@ from io import open
 import tempfile
 from warnings import warn
 
-from .utils import STRING_TYPE, Serialize, SerializeMemoizer, FS, isascii, logger
+from .utils import STRING_TYPE, Serialize, SerializeMemoizer, FS, isascii, logger, ABC, abstractmethod
 from .load_grammar import load_grammar, FromPackageLoader, Grammar
 from .tree import Tree
 from .common import LexerConf, ParserConf
@@ -189,6 +191,14 @@ _LOAD_ALLOWED_OPTIONS = {'postlex', 'transformer', 'use_bytes', 'debug', 'g_rege
 
 _VALID_PRIORITY_OPTIONS = ('auto', 'normal', 'invert', None)
 _VALID_AMBIGUITY_OPTIONS = ('auto', 'resolve', 'explicit', 'forest')
+
+
+class PostLex(ABC):
+    @abstractmethod
+    def process(self, stream):
+        return stream
+
+    always_accept = ()
 
 
 class Lark(Serialize):

--- a/lark/tools/standalone.py
+++ b/lark/tools/standalone.py
@@ -56,7 +56,6 @@ EXTRACT_STANDALONE_FILES = [
     'utils.py',
     'tree.py',
     'visitors.py',
-    'indenter.py',
     'grammar.py',
     'lexer.py',
     'common.py',
@@ -65,6 +64,7 @@ EXTRACT_STANDALONE_FILES = [
     'parsers/lalr_analysis.py',
     'parser_frontends.py',
     'lark.py',
+    'indenter.py',
 ]
 
 def extract_sections(lines):

--- a/lark/utils.py
+++ b/lark/utils.py
@@ -12,6 +12,15 @@ logger.addHandler(logging.StreamHandler())
 # By default, we should not output any log messages
 logger.setLevel(logging.CRITICAL)
 
+if sys.version_info[0]>2:
+    from abc import ABC, abstractmethod
+else:
+    from abc import ABCMeta, abstractmethod
+    class ABC(object): # Provide Python27 compatibility
+        __slots__ = ()
+        __metclass__ = ABCMeta
+
+
 Py36 = (sys.version_info[:2] >= (3, 6))
 
 NO_VALUE = object()

